torch/csrc/jit/codegen/cuda/expr_simplifier.cpp

```diff
@@ -833,14 +941,83 @@ Val* eliminateTrivialComputation(Val* value) {
   return value;
 }
 
+// If x can be proved to be non-negative, then replace x >= 0 as true, replace
+// x < 0 as false
+// If x can be proved to be positive, then replace x >= 0 and x > 0 as true,
+// replace x <= 0 and x < 0 as false
+// If x can be proved to be nonzero, then replace x != 0 as true, replace x == 0
+// as false
+// if x->sameAs(y), then replace x == y as true, replace x != y as false
+Val* eliminateTrivialPredicate(Val* value, const VarInfo& var_info) {
+  if (value->isABool()) {
+    if (auto bop = dynamic_cast<BinaryOp*>(value->definition())) {
+      auto op = bop->getBinaryOpType();
+      if (bop->lhs()->sameAs(bop->rhs())) {
+        if (op == BinaryOpType::Eq) {
+          return IrBuilder::newConstant(true, DataType::Bool);
```

Comment:
Can we use `Fusion::trueVal` and `Fusion::falseVal` instead?

Reply:
fixed